pinot-controller/src/main/java/org/apache/pinot/controller/api/resources/PinotQueryResource.java

```diff
@@ -390,7 +396,9 @@ public String sendPostRaw(String urlStr, String requestStr, Map<String, String>
       /*if (LOG.isInfoEnabled()){
         LOGGER.info("The http response code is " + responseCode);
       }*/
-      if (responseCode != HttpURLConnection.HTTP_OK) {
+      if (responseCode == HttpURLConnection.HTTP_FORBIDDEN) {
```

Comment:
Should we first check response code before getting the output stream?

Reply:
output stream is used to send the payload, input and error streams are read after checking the status code. The logic looks correct.

Reply:
Ah, sorry I misread the code, nvm